### DISCUSSION TITLE
I sent some bitcoin to public master address (Path 'm'), but wallet info not display this

### DISCUSSION THREAD

**Discussion by kernaldpark** [1200wd/bitcoinlib#377] [Q&A] [answered]
1. I created a new wallet with a new private key (tprv8ZgxMBicQKsPeTp7XSVQLBwA5dPJ ******************) - Success
2. And I received some Bitcoin from faucet site to path 93m/44' /1'/0'/0/0 myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF address - Success
3. And I received some Bitcoin from faucet site to path 88 m mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW address - Success
4. I can find at : **https://blockstream.info/testnet/address/mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW**
-> **1 outputs (0.00002300 tBTC)**
![wallet-error000](https://github.com/1200wd/bitcoinlib/assets/11409834/40ea97eb-aca4-4577-9c9f-89dfb699f4cf)

5. However, as shown in the wallet info below, the number of coins for the address 'mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW' is displayed as 0.

**The reason why the number of coins sent to an address with the Path 'm' (Master Public key address) is not displayed, and how to display it, I'm curious about.  If someone knows any solution for anyone of them, I will be thankful for your help**

- my code here
...
keys='tprv8ZgxMBicQKsPeTp7XSVQLBwA5dPJ ******************'
w = wallet_create_or_open(name='today240304wallet1229', keys=keys, network='testnet')
w.transactions_update()
w.utxos_update()
w.scan()
w.info(detail=5)
...
=== WALLET ===
 ID                             10
 Name                           today240304wallet1245
 Owner
 Scheme                         bip32
 Multisig                       False
 Witness type                   legacy
 Main network                   testnet
 Latest update                  2024-03-05 16:18:25.959852

= Wallet Master Key =
 ID                             88
 Private                        True
 Depth                          0

- NETWORK: testnet -
- - Keys
   88 m                            **mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW**            today240304wallet1245               **0.00000000 TBTC**
   89 m/44'                        mniJobrmWoVDfLicq9uA4Hkio44K6hUwRD            purpose 44                          0.00000000 TBTC
   90 m/44'/1'                     myJvzboeMtZJin2zGeYhBYFXNWfe4KGy1N            coin type 1                         0.00000000 TBTC
   91 m/44'/1'/0'                  mguK3d25bvR3ZVsyrPhwnAMicRcUjjhBir            account 0                           0.00000000 TBTC
   96 m/44'/1'/1'                  mtuN1sKSKg2y4qTUvw5Kp43aujdCKc8hRY            Hong Account 1                      0.00000000 TBTC
  101 m/44'/1'/2'                  mqcHqp5g9bpLYAcxoCHGWkBxHPFA8HGVmA            Park Account 1                      0.00000000 TBTC
  106 m/44'/1'/3'                  muP2Uxm6Uq8KuRtbeCc4J9tFRugxDKfZbq            Hong Account 1                      0.00000000 TBTC
  111 m/44'/1'/4'                  myMN7CCvnoP5KSzCSCKb4jSEd8KXqq4pHD            Park Account 1                      0.00000000 TBTC
   92 m/44'/1'/0'/0                mnKr8kEqEE4KRZnge17PNpKzF36dBZdRws            change 0                            0.00000000 TBTC
   94 m/44'/1'/0'/1                mwG6hQ388BETNqkPwVskX3v7BXQ9v1XiHv            change 1                            0.00000000 TBTC
   97 m/44'/1'/1'/0                mozcLDY6w23o14BVE2V9LDRPSjy8Yi85q6            change 0                            0.00000000 TBTC
   99 m/44'/1'/1'/1                mxMJ3ShdjLiZ2WBZJTpaWc4xw6of7J31W1            change 1                            0.00000000 TBTC
  102 m/44'/1'/2'/0                mfr4D6HqFXaUFDjD5NHt1NkJ2NvQLCSMM3            change 0                            0.00000000 TBTC
  104 m/44'/1'/2'/1                mx215ing8AYMyAyYKhHBi4hzuUeCidkUvU            change 1                            0.00000000 TBTC
  107 m/44'/1'/3'/0                n3pvWXMRcoPupTrvBJssxUSDtkUpdwdbqw            change 0                            0.00000000 TBTC
  109 m/44'/1'/3'/1                moe8VNZtfBjSxLVGDncPePxYDg6a4HLHah            change 1                            0.00000000 TBTC
  112 m/44'/1'/4'/0                mxTLPEhdnsYvrcvRgBqBQLxTgnvd7UugWh            change 0                            0.00000000 TBTC
  114 m/44'/1'/4'/1                mt5YSCTBayHq1MmvJw3PFwncNYwu5fCxQt            change 1                            0.00000000 TBTC
   93 m/44'/1'/0'/0/0              myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF            address index 0                     0.00000000 TBTC
   95 m/44'/1'/0'/1/0              n1eygexyyAy9DDsifdkP36JUiWwv7UoKeF            address index 0                     0.00000000 TBTC
   98 m/44'/1'/1'/0/0              mnszxkBDa7h6QQ1DyqiDSXhtYSbWUAnim8            address index 0                     0.00000000 TBTC
  100 m/44'/1'/1'/1/0              myUsAv7smjC89gXvMRFfQiFjcDKbAHUqeQ            address index 0                     0.00000000 TBTC
  103 m/44'/1'/2'/0/0              mqEKrHK2ZJ4Qh4KSyzAuV1xYyanMQkhYT6            address index 0                     0.00000000 TBTC
  105 m/44'/1'/2'/1/0              n3EQq6bXUTJJpc7Rx7Gc8hoEiL1sht7ZKg            address index 0                     0.00000000 TBTC
  108 m/44'/1'/3'/0/0              mt3Lj91fpgDBy3VBWZKBZXsxgRrPDYxAyx            address index 0                     0.00000000 TBTC
  110 m/44'/1'/3'/1/0              mjB81NWsQcLmtw2T2vtsUSzKrEg45B9SbR            address index 0                     0.00000000 TBTC
  113 m/44'/1'/4'/0/0              muWarfoShq8WybCefoMoD4DtXGgyd96jDn            address index 0                     0.00000000 TBTC
  115 m/44'/1'/4'/1/0              mztfsy2YXz73P3WJaBEUbvDfsM8E7v1Mh6            address index 0                     0.00000000 TBTC
  163 m/44'/1'/0'/0/1              mzDEmFdvkVKEFB6GqpQsBGktLETX7VBAbJ            address index 1                     0.00000000 TBTC
  164 m/44'/1'/0'/0/2              mm99anV1nT17Gchgs12u8XWjaYRw1J2N1z            address index 2                     0.00000000 TBTC
  165 m/44'/1'/0'/0/3              mfxPuhsHaoEBER15YhHY8nUAeJSZcrWsYg            address index 3                     0.00000000 TBTC
  166 m/44'/1'/0'/0/4              mthve42ba21qwhMBctka2YSXwXPKSE5uSB            address index 4                     0.00000000 TBTC
  167 m/44'/1'/0'/0/5              mhvZNHkXnWfNgTJzvFCY9eLD2QUtPj4XbM            address index 5                     0.00000000 TBTC
  168 m/44'/1'/0'/0/6              n2v56jQsXHpJbCcLJ8r63YY5qqygAfcxC1            address index 6                     0.00000000 TBTC
  169 m/44'/1'/0'/1/1              mgtJszmihyfBGMZKb5RbtCBGzAh4CxwVnk            address index 1                     0.00000000 TBTC
  170 m/44'/1'/0'/1/2              mqQAuAPWjzGUkFRyipPbBNRhXbn97nydqj            address index 2                     0.00000000 TBTC
  171 m/44'/1'/0'/1/3              mu68fA9wyecHgk4YYtoo9TD6ubFWgJpxz7            address index 3                     0.00000000 TBTC
  172 m/44'/1'/0'/1/4              msxUYKHDSC6B3scYV6SGULSGrrkzNwFJLV            address index 4                     0.00000000 TBTC

- - Transactions Account 0 (12)
e4389b42076dc4906174ac46c1f38a67ef6f9a15180818b711f22601388a49a5          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      138       0.00002000 TBTC
7ad5595431418db06d3fda5b2b1257a1cfc86a167804528092bc68d6446ae5c0          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      136       0.00001500 TBTC
9cec7c8be84079b0b9210ecd695dc1f6deb7e82c8c249e3bd5a1175a15b23ebc          mzDEmFdvkVKEFB6GqpQsBGktLETX7VBAbJ      134       0.00001500 TBTC
232210ec0fddd2179632588020e5547bf1d0fc9505d1421ad3e052f36adf971f          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      132      -0.00002000 TBTC
b485484e34ea4ace32e73fa97984b42eafaca1e167b9a500c7863b403ca06390          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      132      -0.00001500 TBTC
5b1bb2c07982565cd049c8f95086d7d928abf0c4964a7ad8239b439d6f3cbfbe          mzDEmFdvkVKEFB6GqpQsBGktLETX7VBAbJ      131      -0.00001500 TBTC
a3d069dff95bff033fe94048ff4d4f07557039828a5b37cb4914a0c75c4b308e          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      130       0.00001300 TBTC
e7354b769e693658e947da4ee7972ec22898a93b466926f0fa218c307275d9f4          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      106      -0.00001300 TBTC
a377284e4163562e2a91ae34c9d67fa25d4c60ac9ae88a1d18df00b7f9e149fb          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF       99       0.00002000 TBTC
42666e9db48b9732a9ce570fb1e2f6adc5768cbff3e122c03df3320439d5256d          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF       96      -0.00002000 TBTC
42666e9db48b9732a9ce570fb1e2f6adc5768cbff3e122c03df3320439d5256d          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF       96       0.00001700 TBTC
44a460157d3e090629ec71de47cfc3a9d3f215769315f8c76e613c367331324c          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF       93      -0.00001700 TBTC

- - Transactions Account 1 (0)

- - Transactions Account 2 (0)

- - Transactions Account 3 (0)

- - Transactions Account 4 (0)

= Balance Totals (includes unconfirmed) =

**Comment by mccwdev** [comment, marked as answer]
Normally you would only use the top level addresses with depth 5, the rest of the key path is not used for receiving and sending.
To get a address for receiving funds you can use the get_key method:

``` python
w.get_key().address
```

However you can update the master key address by specifying the depth when updating the transactions:

``` python
w.transactions_update(depth=0)
```

**Comment by kernaldpark** [comment]
Thank you very much!
-> w.transactions_update(depth=0)

=== WALLET ===
 ID                             9
 Name                           today240304wallet1229     
 Owner
 Scheme                         bip32
 Multisig                       False
 Witness type                   legacy
 Main network                   testnet
 Latest update                  2024-03-10 22:21:16.868645

= Wallet Master Key =
 ID                             80
 Private                        True
 Depth                          0

- NETWORK: testnet -
- - Keys
   80 m                            mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW            today240304wallet1229               0.00000000 TBTC
   81 m/44'                        mniJobrmWoVDfLicq9uA4Hkio44K6hUwRD            purpose 44                          0.00000000 TBTC
   82 m/44'/1'                     myJvzboeMtZJin2zGeYhBYFXNWfe4KGy1N            coin type 1                         0.00000000 TBTC
   83 m/44'/1'/0'                  mguK3d25bvR3ZVsyrPhwnAMicRcUjjhBir            account 0                           0.00000000 TBTC
   84 m/44'/1'/0'/0                mnKr8kEqEE4KRZnge17PNpKzF36dBZdRws            change 0                            0.00000000 TBTC
   86 m/44'/1'/0'/1                mwG6hQ388BETNqkPwVskX3v7BXQ9v1XiHv            change 1                            0.00000000 TBTC
   85 m/44'/1'/0'/0/0              myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF            address index 0                     0.00000000 TBTC
   87 m/44'/1'/0'/1/0              n1eygexyyAy9DDsifdkP36JUiWwv7UoKeF            address index 0                     0.00000000 TBTC
  143 m/44'/1'/0'/0/1              mzDEmFdvkVKEFB6GqpQsBGktLETX7VBAbJ            address index 1                     0.00000000 TBTC
  144 m/44'/1'/0'/0/2              mm99anV1nT17Gchgs12u8XWjaYRw1J2N1z            address index 2                     0.00000000 TBTC
  145 m/44'/1'/0'/0/3              mfxPuhsHaoEBER15YhHY8nUAeJSZcrWsYg            address index 3                     0.00000000 TBTC
  146 m/44'/1'/0'/0/4              mthve42ba21qwhMBctka2YSXwXPKSE5uSB            address index 4                     0.00000000 TBTC
  147 m/44'/1'/0'/0/5              mhvZNHkXnWfNgTJzvFCY9eLD2QUtPj4XbM            address index 5                     0.00000000 TBTC
  148 m/44'/1'/0'/1/1              mgtJszmihyfBGMZKb5RbtCBGzAh4CxwVnk            address index 1                     0.00000000 TBTC
  149 m/44'/1'/0'/1/2              mqQAuAPWjzGUkFRyipPbBNRhXbn97nydqj            address index 2                     0.00000000 TBTC
  150 m/44'/1'/0'/1/3              mu68fA9wyecHgk4YYtoo9TD6ubFWgJpxz7            address index 3                     0.00000000 TBTC
  151 m/44'/1'/0'/1/4              msxUYKHDSC6B3scYV6SGULSGrrkzNwFJLV            address index 4                     0.00000000 TBTC
  153 m/44'/1'/0'/0/6              n2v56jQsXHpJbCcLJ8r63YY5qqygAfcxC1            address index 6                     0.00000000 TBTC

- - Transactions Account 0 (15)
e4389b42076dc4906174ac46c1f38a67ef6f9a15180818b711f22601388a49a5          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      845       0.00002000 TBTC
7ad5595431418db06d3fda5b2b1257a1cfc86a167804528092bc68d6446ae5c0          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      843       0.00001500 TBTC
9cec7c8be84079b0b9210ecd695dc1f6deb7e82c8c249e3bd5a1175a15b23ebc          mzDEmFdvkVKEFB6GqpQsBGktLETX7VBAbJ      841       0.00001500 TBTC
232210ec0fddd2179632588020e5547bf1d0fc9505d1421ad3e052f36adf971f          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      839      -0.00002000 TBTC
b485484e34ea4ace32e73fa97984b42eafaca1e167b9a500c7863b403ca06390          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      839      -0.00001500 TBTC
5b1bb2c07982565cd049c8f95086d7d928abf0c4964a7ad8239b439d6f3cbfbe          mzDEmFdvkVKEFB6GqpQsBGktLETX7VBAbJ      838      -0.00001500 TBTC
a3d069dff95bff033fe94048ff4d4f07557039828a5b37cb4914a0c75c4b308e          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      837       0.00001300 TBTC
71713319c196fbd1a3548cf6065d1dd4f1036bc865ada2d81e7bed34ecde8405          mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW      837       0.00001500 TBTC
e7354b769e693658e947da4ee7972ec22898a93b466926f0fa218c307275d9f4          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      813      -0.00001300 TBTC
55859d2670af2b1847edc40787b912ffbea6851513eb0c9382f3fdf8031b233e          mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW      813      -0.00001500 TBTC
a377284e4163562e2a91ae34c9d67fa25d4c60ac9ae88a1d18df00b7f9e149fb          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      806       0.00002000 TBTC
42666e9db48b9732a9ce570fb1e2f6adc5768cbff3e122c03df3320439d5256d          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      803      -0.00002000 TBTC
42666e9db48b9732a9ce570fb1e2f6adc5768cbff3e122c03df3320439d5256d          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      803       0.00001700 TBTC
44a460157d3e090629ec71de47cfc3a9d3f215769315f8c76e613c367331324c          myQqsb7ZpGViEFGQuCiNQtu5ahjntr2vhF      800      -0.00001700 TBTC
350f01088bb3b99ba41d718ba99bd83c525810dd6dcfac3d556b0669113e8f5a          mqqe9TV41qLSFvvzydFu3VoTYNjoU8N2xW      708       0.00002300 TBTC

= Balance Totals (includes unconfirmed) =
**testnet              (Account 0)               0.00002300 TBTC**
